src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -158,7 +171,8 @@ def run(
                     # set batch input source from input mapping
                     OperationContext.get_instance().set_batch_input_source_from_inputs_mapping(inputs_mapping)
                     # resolve input data from input dirs and apply inputs mapping
-                    batch_input_processor = BatchInputsProcessor(self._working_dir, self._flow.inputs, max_lines_count)
+                    inputs = self._flow.inputs if self._flow else self._executor_proxy.get_inputs_definition()
```

Comment:
Add some comments here about the if/else logic

Reply:
Added.